src/cljs/rems/fields.cljs

```diff
@@ -169,13 +169,18 @@
   (.. event -target -value))
 
 (defn text-field
-  [{:keys [validation on-change info-text] :as opts}]
+  [{:keys [validation on-change info-text type] :as opts}]
   (let [placeholder (localized (:field/placeholder opts))
         value (:field/value opts)
         optional (:field/optional opts)
-        max-length (:field/max-length opts)]
+        max-length (:field/max-length opts)
+        type (:field/type opts)]
     [field-wrapper opts
-     [:input.form-control {:type "text"
+     [:input.form-control {:type (cond
+                                   (= type :text) "text"
+                                   (= type :phone-number) "tel"
+                                   (= type :email) "email"
+                                   :else "text")
```

Comment:
Use `case`? instead of this `cond`?

I think these are piggybacking a bit too much on the `text-field` function already. We could rename `text-field` to `basic-input-field` or something?

Reply:
I agree, case looks like a better fit here, also the text-field is misleading as we do not only put text there, Ill try to think of a better name.

Reply:
Also, @Macroz I am wondering how much date-field is just a repetition of text-field but with date?
Do you think it makes sense to unify them? Create one universal function for input field?

Reply:
Seems to me like date-field could indeed be unified. Might be a good idea to do it in a separate PR though.

Reply:
ok, but good action point

Reply:
I ll create a small ticket

Reply:
There doesn't need to be a new ticket btw. It can be a new bullet point to the current ticket since this is for it still.